### ISSUE TITLE
Show self-limiting commitments

### ISSUE THREAD

**Issue by micho** [StakingDirectory/StakingDirectory#57]
From https://twitter.com/superphiz/status/1697035864793071887:

> @superphiz: These providers are committed (or are in the process of committing) to self-limit to <22% of Ethereum validators. This is how our chain will be successful: Coordination above greed. Cooperation instead of winner-take-all.
> 
> @Rocket_Pool
> @stakewise_io
> @staderlabs
> @divastaking

This would be great to surface! It could also be linked to the details for each one:

- Rocketpool has https://github.com/rocket-pool/RPIPs/blob/main/RPIPs/RPIP-17.md
- For Stakewise and Stader I couldn't find DAO resolutions, but that might be my bad, will keep searching
- Diva is preparing a DAO vote to self limit

@EridianAlpha @Spacesider do you have any wishes or ideas on how to implement this? Would be happy to prepare a PR, but wanted to ask first :)

**Comment by EridianAlpha** [comment]
Thanks for raising this @micho 🙂 I will add this as a new data field, and then look at the the best place to display it.

**Comment by superphiz** [comment]
Stader: https://blog.staderlabs.com/staders-ethx-power-pledge-our-promise-to-keep-ethereum-decentralized-c482ec66641c?gi=5e6bb767fd2b

**Comment by superphiz** [comment]
Stakewise: https://twitter.com/stakewise_io/status/1525225299146944513

**Comment by superphiz** [comment]
Swell: https://twitter.com/swellnetworkio/status/1697441725567119487

**Comment by superphiz** [comment]
Puffer: https://twitter.com/StecoinZhang/status/1697461300258046232

**Comment by EridianAlpha** [comment]
Thanks for all these links @superphiz 🙏

**Comment by superphiz** [comment]
StaFi Protocol: https://twitter.com/StaFi_Protocol/status/1697560727664709798?s=20

**Comment by EridianAlpha** [comment]
> Puffer: https://twitter.com/StecoinZhang/status/1697461300258046232

Puffer hasn't been added to Staking Directory yet, so I'll ask them to add themselves, then this self-limit can be included.

**Comment by EridianAlpha** [comment]
Here are the initial designs @superphiz @micho 
![image](https://github.com/StakingDirectory/StakingDirectory/assets/86061486/aa8e0bc5-2a4d-482c-99c7-48256795e0db)

**Comment by superphiz** [comment]
I wonder what it would look like to split that box horizontally and on the left have "network share" and "self limit commitment" on the right (?)

**Comment by micho** [comment]
i like that idea! this dashboard has good data sources https://dune.com/hildobby/eth2-staking

if data isnt added yet, this is a good way to visualize the self limit

**Comment by EridianAlpha** [comment]
@micho @superphiz This is an initial design, I'm going to work on styling and colors so it's clear when something is good or bad for the network.

I'm going to pull the data from the rated.network API so that it remains current. I'll run a GitHub action every 24 hours to update the data.

![image](https://github.com/StakingDirectory/StakingDirectory/assets/86061486/48a41945-8239-4989-bcb4-e566f8b74c66)

**Comment by micho** [comment]
looks great! i like that its inside and not in the list, because placing market share in the list would make it look like bigger is better when scanning through the options

**Comment by EridianAlpha** [comment]
@superphiz @micho this feature is now live and will automatically update the values every 24 hours.

When there's no rated.network ID set, the value used is the 30 day average APR for the entire Ethereum network. I think this is helpful for options like Dappnode that don't have their own rated.network operator entry, so just show the entire network average.

**Comment by micho** [comment]
amazing!! thanks for the super fast delivery time @EridianAlpha 

i agree on the APR for those without values. as an improvement, it could be listed as a gray value with a tooltip or (*) saying its the network average

**Comment by micho** [comment]
next up, it seems like a diva dao vote is coming to confirm the self limit, so ill link to that once there's a decision!

**Comment by EridianAlpha** [comment]
> i agree on the APR for those without values. as an improvement, it could be listed as a gray value with a tooltip or (*) saying its the network average

@micho so I did play around with this when I was designing it, but making the network average values gray made them look "less good" than the LSTs. So I went with the border around the LSTs and no border around the network average, just so there is some visible difference. A * for the network average values could look good though, as it wouldn't make them look "worse" but would show that there's something different about those values.
